init.lua

```diff
@@ -197,131 +197,133 @@ local function smooth(pos1, pos2, deadzone)
 	local offset = vector.subtract(pos1, area.MinEdge)
 	local c_air = minetest.get_content_id("air")
 	local c_dirt = minetest.get_content_id("default:dirt")
-
-	-- read heightmap from data
-	local heightmap = {}
-	local hstride = {x=1, z=dim.x}
-	for x = 0, dim.x-1 do
-		for z = 0, dim.z-1 do
-			heightmap[x + (z * hstride.z) + 1] = 0
+	
+	local count = 0
+	for i = 1,iterations do
+		-- read heightmap from data
```

Comment:
reading/writing the map only needs to be done once

Reply:
Ah, you're right. I didn't notice that! I'll move the heightmap calculation outside the loop.